### DISCUSSION TITLE
Upgrade Question

### DISCUSSION THREAD

**Discussion by sdresen** [pfelk/pfelk#278] [Q&A] [answered]
When I installed pfElk I didn't have Snort enabled at the time and thus didn't load the Snort json file.  I've since started using Snort again and went to load the current Snort json file but am getting the following error:

Sorry, there was an error

The file could not be processed due to error: "Unprocessable Entity: Document "02445e60-116b-11eb-8725-ed0ed9b0ff36" has property "visualization" which belongs to a more recent version of Kibana [7.11.0]. The last known version is [7.10.0]"

Was hunting for instructions in your docs about how to upgrade but haven't found them yet.  Can you point me to them or let me know if there is a way to upgrade?

Thanks

**Comment by a3ilson** [comment, marked as answer]
@sdresen - Sorry for the delay... I do not always get notifications from the discussions.

So to clarify, the issue you are having is that when attempting to import the snort.ndjson file, you are prompted with an error identified above? Seems odd...I do not run snort but was able to import within my current instance (7.11.0).

**Reply by sdresen** [reply]
Correct, the issue is when I attempt to import the snort.ndjson.  You not getting the error makes sense to me as the error suggests the issue is that I'm running 7.10.0 and need to be running 7.11.0 like you are for the import to work.  So, I was wondering what is the best way to upgrade my pfElk environment.  Whether now or later, at some point I'd likely need/want to upgrade and am not sure the best way to do that.

**Reply by a3ilson** [reply]
Gotcha - Well this depends, if you do not want any downtime, I can help with finding a prior versions of snort.

However, upgrading Elastic is simple - how do you have it installed? I utilize Docker for mine which allows for a separation from my host while allowing simple upgrades. However, if it is installed on your host, simply upgrade (`sudo apt-get update; sudo -apt-get upgrade; dist-upgrade; sudo apt-get`). This is another reason I elect to utilize containers, other host related upgrades may be isolated (i.e. security). Once upgrade, you may need to restart the services/host and you'll be back up and running.

**Reply by sdresen** [reply]
Downtime is not a problem as this is a hobby project for me.  I used your install script to set the system up and believe everything is running in containers.  I didn't want to start poking around trying things at the risk of not knowing how your script set everything up.

**Reply by a3ilson** [reply]
Too easy.  The scripted install simply automates the manual installation for Ubuntu/Debian. So running a simple `sudo apt-get update; sudo -apt-get upgrade; dist-upgrade; sudo apt-get` will also upgrade Elastic and after restarting, you'll be back up and running.

A few pointers on the scripted install or in general. If you have it installed on a VM - you can take a snapshot. This will allow you to tinker and when/if you break something, you can simply revert back to the working snapshot.

I would also take a look at [this](https://github.com/pfelk/pfelk/wiki/How-To:-Performance) as you may want to adjust your memory settings to tweak performance with logstash/elasticsearch.

**Reply by sdresen** [reply]
Thanks for the pointers.  I am running it in a VM and took a snapshot so I could try a few things and revert back as needed.  In doing that I realized I was running with docker containers, not the scripted install.  I ended up exporting my visualizations from the instance I was running and just rerunning your script so I was using pfElk natively on the OS versus containers given I have a dedicated VM for it.  That way, future updates will be automatic as I run OS/package updates.  As I'm still learning, I wasn't able to figure out how to update the docker version of pfElk with existing data.  I was able to download updated ELK images, changed the tags of the images, but every time I ran docker compose, the containers rebuilt with the same version I had been running on instead of the updated images I had downloaded.

**Reply by a3ilson** [reply]
You'll need to edit the docker-compose.yml file and amend the version and/or variable.

**Reply by sdresen** [reply]
I was on that path having determined in docker-compose.yml that the following was driving the version it was using for ELK

args:
         ELK_VERSION: $ELK_VERSION_

In the Dockerfile for each component, the 

ARG ELK_VERSION
FROM docker.elastic.co/elasticsearch/elasticsearch:${ELK_VERSION}

looks to be pulling version from the image source.  I got stuck at this point trying to figure out what to modify to get it to use the updated image I had pulled down.

**Reply by a3ilson** [reply]
there should be a .env file (`nano .env`) and amend the version.  Then restart the docker containers.

**Reply by sdresen** [reply]
So close, thanks.
